### PR TITLE
fix dialog (imperative) examples to use observables

### PR DESCRIPTION
**Pull request by andrew-bents**
Imperative dialogs examples in demo app use promise based implementation of dialog service. Submitting the pull request to change it to observables

### CODE REVIEW

File: src/demo-app/app/dialog/dialog.component.html [hunk 1]
```diff
@@ -202,18 +223,18 @@ <h3 class="mdl-dialog__title">App Login</h3>
...
 <scripts>
-  public showDialog() {
-
+  public showDialog($event: MouseEvent) {
     let pDialog = this.dialogService.showCustomDialog({
       component: LoginDialogComponent,
       providers: [{provide: TEST_VALUE, useValue: 'Just an example'}],
       isModal: true,
-      styles: {'width': '350px'},
+      styles: {'width': '300px'},
```
Comment (mseemann) [inline src/demo-app/app/dialog/dialog.component.html]: why did you change the dialog width?

File: src/demo-app/app/dialog/dialog.component.html [hunk 2]
```diff
@@ -202,18 +223,18 @@ <h3 class="mdl-dialog__title">App Login</h3>
...
       clickOutsideToClose: true,
+      openFrom: $event,
```
Comment (mseemann) [inline src/demo-app/app/dialog/dialog.component.html]: why did you add the option openFrom?

**Comment by mseemann** [comment]
Hi thx for your contribution and fixing the typos :). Can you answer my two small questions? thx.

File: src/demo-app/app/dialog/dialog.component.html (hunk 1 above)
Comment (andrew-bents) [inline src/demo-app/app/dialog/dialog.component.html]: well, i copied that dialog configuration from the actual code (https://git.io/vyl3u)

File: src/demo-app/app/dialog/dialog.component.html (hunk 2 above)
Comment (andrew-bents) [inline src/demo-app/app/dialog/dialog.component.html]: and again, took it from the code. I can remove it if you want

**Comment by mseemann** [comment]
ok. i see. i'll change the two lines manually. thx!

**Comment by andrew-bents** [comment]
Thanks for merging! But i think the github pages hasn't been updated and show the old docs :)

**Comment by mseemann** [comment]
I have not updated the page yet because I want to finish the work for the next version.  Will be visible in a few days....
